src/nomad_simulations/schema_packages/general.py

```diff
@@ -262,19 +266,84 @@ def set_composition_formula(system_parent: ModelSystem) -> None:
         for child in system_parent.sub_systems:
             Simulation.set_composition_formula(system_parent=child)
 
+    @log
+    def _validate_and_set_representative_system(self) -> None:
+        """
+        Ensure exactly one representative `ModelSystem` exists in `self.model_system`,
+        and update `self.representative_system_index` to point to it.
+
+        Behavior
+        --------
+        - **Zero representatives:** Mark the *last* item in `self.model_system` as
+        representative and set `self.representative_system_index = -1`
+        (Python’s negative index for the last element).
+        - **Multiple representatives:** Log a warning, demote all, then keep the
+        *last* representative encountered. Store its (non-negative) index in
+        `self.representative_system_index`.
```

Comment:
This works under the presumption that the model_systems follow an order...
Maybe put it as a disclaimer? Either at the documentation level or the programmtic level, we should enforce this.

Reply:
Simulation.model_systems is a list, so it has an order. I don't understand why it would be unclear, can you explain please?

Reply:
> Simulation.model_systems is a list, so it has an order.

Lists aren't necessarily ordered, e.g. `[2, 3, 1]`. We presume that when a natural ordering (let's say iteration steps or time) is present, it is applied to `model_systems`, but this assumption is always implicit and not enforced. This is a good opportunity for making it explicit, showing why it's important.